apis/network/v1alpha3/resourcerecordset_types.go

```diff
@@ -0,0 +1,137 @@
+package v1alpha3
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSRecordType defines the valid DNS Record Types that can be used.
+type DNSRecordType string
+
+const (
+	// DNSRecordTypeSOA represents DNS SOA record type.
+	DNSRecordTypeSOA DNSRecordType = "SOA"
+
+	// DNSRecordTypeA represents DNS A record type.
+	DNSRecordTypeA DNSRecordType = "A"
+
+	// DNSRecordTypeTXT represents DNS TXT record type.
+	DNSRecordTypeTXT DNSRecordType = "TXT"
+
+	// DNSRecordTypeNS represents DNS NS record type.
+	DNSRecordTypeNS DNSRecordType = "NS"
+
+	// DNSRecordTypeCNAME represents DNS CNAME record type.
+	DNSRecordTypeCNAME DNSRecordType = "CNAME"
+
+	// DNSRecordTypeMX represents DNS MX record type.
+	DNSRecordTypeMX DNSRecordType = "MX"
+
+	// DNSRecordTypeNAPTR represents DNS NAPTR record type.
+	DNSRecordTypeNAPTR DNSRecordType = "NAPTR"
+
+	// DNSRecordTypePTR represents DNS PTR record type.
+	DNSRecordTypePTR DNSRecordType = "PTR"
+
+	// DNSRecordTypeSRV represents DNS SRV record type.
+	DNSRecordTypeSRV DNSRecordType = "SRV"
+
+	// DNSRecordTypeSPF represents DNS SPF record type.
+	DNSRecordTypeSPF DNSRecordType = "SPF"
+
+	// DNSRecordTypeAAAA represents DNS AAAA record type.
+	DNSRecordTypeAAAA DNSRecordType = "AAAA"
+
+	// DNSRecordTypeCAA represents DNS CAA record type.
+	DNSRecordTypeCAA DNSRecordType = "CAA"
+)
+
+// ChangeAction defines the valid actions that can be performed on a ResourceRecordSet.
+type ChangeAction string
+
+const (
+	// ChangeActionCreate represents a Resource Record CREATE operation.
+	ChangeActionCreate ChangeAction = "CREATE"
+
+	// ChangeActionDelete represents a Resource Record DELETE operation.
+	ChangeActionDelete ChangeAction = "DELETE"
+
+	// ChangeActionUpsert represents a Resource Record UPSERT operation.
+	ChangeActionUpsert ChangeAction = "UPSERT"
+)
+
+// +kubebuilder:object:root=true
+
+// ResourceRecordSet is a managed resource that represents an AWS Route53 Resource Record.
+// +kubebuilder:printcolumn:name="TYPE",type="string",JSONPath=".spec.forProvider.type"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type ResourceRecordSet struct {
```

Comment:
I believe the name should be `ResourceRecord`. The `List` and `Change` calls use `ResourceRecordSet` name but that's because they work as batch operations. Here, we expose individual resource record as CR.

Reply:
The name is `ResourceRecordSet` because we aren't exposing individual Resource Record but rather a bunch/group  of those together as ```
Records []string `json:"records"`
``` in the `ResourceRecordParameters`